src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java

```diff
@@ -85,6 +94,13 @@ public ReportsService(
     }
 
     private ZipFileSummaryResponse fromDbZipfileSummary(ZipFileSummary dbItem) {
+        List<PaymentResponse> payments;
+        if (dbItem.getEnvelopeId() == null) {
+            payments = emptyList();
+        } else {
+            payments = toPaymentsResponse(
+               envelopeRepo.findById(UUID.fromString(dbItem.getEnvelopeId())).get().getPayments());
```

Comment:
can we add payment details to existing query. in `ZipFilesSummaryRepository` so we'd not need extra query and 'envelopeRepo'  ?

Reply:
I tried it but it didn't work with another join in the query